src/ripple/app/ledger/impl/InboundLedgers.cpp

```diff
@@ -67,26 +66,26 @@ class InboundLedgersImp
                 auto it = mLedgers.find (hash);
                 if (it != mLedgers.end ())
                 {
-                    // Don't touch failed acquires so they can expire
-                    if (! it->second->isFailed ())
-                    {
-                        it->second->update (seq);
-                        if (it->second->isComplete ())
-                            ret = it->second->getLedger ();
-                    }
+                    inbound = it->second;
 
+                    // If the acquisition failed, don't mark the item as
+                    // recently accessed so that it can expire.
+                    if (! inbound->isFailed ())
+                        it->second->update (seq);
                 }
                 else
                 {
-                    auto il = std::make_shared <InboundLedger> (hash, seq, reason, std::ref (m_clock));
-                    mLedgers.insert (std::make_pair (hash, il));
-                    il->init (sl);
+                    inbound = std::make_shared <InboundLedger> (
+                        hash, seq, reason, std::ref (m_clock));
+                    mLedgers.emplace (hash, inbound);
```

Comment:
`std::move (inbound)` saves, I believe, one increment and one decrement of the reference counter...